sky/backends/backend_utils.py

```diff
@@ -1308,15 +1308,16 @@ def get_node_ips(cluster_yaml: str,
     ray_config = common_utils.read_yaml(cluster_yaml)
     use_tpu_vm = ray_config['provider'].get('_has_tpus', False)
     if use_tpu_vm:
-        assert expected_num_nodes == 1, (
-            'TPU VM only supports single node for now.')
         assert handle is not None, 'handle is required for TPU VM.'
         try:
-            ips = _get_tpu_vm_pod_ips(ray_config, get_internal_ips)
+            ips = _get_tpu_vm_pod_ips(ray_config, expected_num_nodes,
+                                      get_internal_ips)
         except exceptions.CommandError as e:
             raise exceptions.FetchIPError(
                 exceptions.FetchIPError.Reason.HEAD) from e
-        if len(ips) != tpu_utils.get_num_tpu_devices(handle.launched_resources):
+        per_tpu_devices = tpu_utils.get_num_tpu_devices(
```

Comment:
Good point. I rename it to `num_workers_per_tpu_type` because it can be a TPU VM or Pod.
For `tpu_utils.get_num_tpu_devices`, I name it based on the GCP's official terminology [doc](https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu_v2).
```
A TPU Pod - a group of TPU devices connected by high-speed interconnects
```
I think that's fine? I added some docstring for the function. Does this look clear now?
https://github.com/skypilot-org/skypilot/pull/1185/files#diff-5386aeeded1377e4dc0ab8795064796e3971cbf2d58892ac3ed2161bac8e765fR34